src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -3355,24 +3418,19 @@ namespace detail
             {
                 m_alreadyEnqueued.emplace_back(std::move(task));
             }
-            if (writeAttributes)
+            if (writeLatePuts)
             {
-                for (auto &task : m_attributeWrites)
-                {
-                    m_alreadyEnqueued.emplace_back(
-                        std::unique_ptr<BufferedAction>{
-                            new BufferedAttributeWrite{
-                                std::move(task.second)}});
-                }
-                m_attributeWrites.clear();
+                throw error::Internal(
+                    "ADIOS2 backend: Flush of late writes was requested at the "
+                    "wrong time.");
```

Comment:
This error message is not hard to understand for me :)

Reply:
That is great!

Nonetheless, for us (devs) to remember, I would extend the half sentence about the "wrong time" a bit :D